call of overloaded atan2(0, -1) or exp(int) is ambiguous

With g++ on solaris, compiling symengine results the following error.

``` 
/tmp/symengine.Rcheck/00_pkg_src/symengine/src/upstream/symengine/eval_double.cpp:686:33: 
error: call of overloaded ‘atan2(int, int)’ is ambiguous
                return ::atan2(0, -1);
                                    ^
 
/tmp/symengine.Rcheck/00_pkg_src/symengine/src/upstream/symengine/eval_double.cpp:688:27: 
error: call of overloaded ‘exp(int)’ is ambiguous
                return ::exp(1);
                              ^
```

Guess it might be fixed by passing variable instead of literal, like:

```
int x = 1;
return ::exp(1);
```

Hi @isuruf , I am sorry for the late response. I tried to reproduce this in a solaris VM and attached the full logs below. It seems that it can be fixed with the following patch:

```
diff --git a/symengine/eval_double.cpp b/symengine/eval_double.cpp
index 92c8c59..daa8f23 100644
--- a/symengine/eval_double.cpp
+++ b/symengine/eval_double.cpp
@@ -678,9 +678,9 @@ std::vector<fn> init_eval_double()
     };
     table[SYMENGINE_CONSTANT] = [](const Basic &x) {
         if (eq(x, *pi)) {
-            return ::atan2(0, -1);
+            return std::atan2(0, -1);
         } else if (eq(x, *E)) {
-            return ::exp(1);
+            return std::exp(1);
         } else if (eq(x, *EulerGamma)) {
             return 0.5772156649015328606065; // use until polygamma or digamma
                                              // is implemented
diff --git a/symengine/tests/basic/test_parser.cpp b/symengine/tests/basic/test_parser.cpp
index fb42d39..29fd4d4 100644
--- a/symengine/tests/basic/test_parser.cpp
+++ b/symengine/tests/basic/test_parser.cpp
@@ -629,7 +629,7 @@ TEST_CASE("Parsing: doubles", "[parser]")
     res = parse(s);
     REQUIRE(is_a<RealDouble>(*res));
     d = down_cast<const RealDouble &>(*res).as_double();
-    REQUIRE(std::abs(d - (::sqrt(2) + 5)) < 1e-12);
+    REQUIRE(std::abs(d - (std::sqrt(2) + 5)) < 1e-12);

     // Test that https://github.com/symengine/symengine/issues/1413 is fixed
```

cmake output:

```
-- The C compiler identification is GNU 5.2.0
-- The CXX compiler identification is GNU 5.2.0
-- Check for working C compiler: /opt/csw/bin/gcc
-- Check for working C compiler: /opt/csw/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /opt/csw/bin/g++
-- Check for working CXX compiler: /opt/csw/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test NO_WARN_IMPLICIT_FALL_THROUGH_FLAG
-- Performing Test NO_WARN_IMPLICIT_FALL_THROUGH_FLAG - Success
-- Linker does not support --exclude-libs
-- Linker does not support -unexported_symbol
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of long double
-- Check size of long double - done
-- Found GMP: /opt/csw/lib/libgmp.so
-- cotire 1.7.8 loaded.
-- CXX target symengine cotired.
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
--------------------------------------------------------------------------------

+-------------------------------+
| Configuration results SUMMARY |
+-------------------------------+

CMAKE_C_COMPILER:   /opt/csw/bin/gcc
CMAKE_CXX_COMPILER: /opt/csw/bin/g++
CMAKE_BUILD_TYPE: Release
CMAKE_CONFIGURATION_TYPES:
CMAKE_C_FLAGS:           -O2
CMAKE_CXX_FLAGS:         -std=c++11 -O2 -fPIC
CMAKE_CXX_FLAGS_DEBUG:   -Wall -Wextra -Wno-unused-parameter -fno-common -g -ggdb -Wno-unknown-pragmas
CMAKE_CXX_FLAGS_RELEASE: -Wall -Wextra -Wno-unused-parameter -fno-common -O3 -funroll-loops -Wno-unknown-pragmas
CMAKE_INSTALL_PREFIX: /usr/local
BUILD_SHARED_LIBS:
INTEGER_CLASS : GMP
HAVE_SYMENGINE_GMP: yes
WITH_SYMENGINE_ASSERT: no
WITH_SYMENGINE_RCP: yes
WITH_SYMENGINE_TEUCHOS: no
WITH_COTIRE: yes
WITH_GENERATE_PARSER: no
HAVE_GCC_ABI_DEMANGLE: yes
HAVE_C_FUNCTION_NOT_FUNC: no
HAVE_DEFAULT_CONSTRUCTORS: yes
HAVE_SYMENGINE_NOEXCEPT: yes
HAVE_SYMENGINE_IS_CONSTRUCTIBLE: yes
HAVE_SYMENGINE_RESERVE: yes
HAVE_SYMENGINE_STD_TO_STRING: TRUE
WITH_SYMENGINE_THREAD_SAFE: no
BUILD_TESTS: yes
BUILD_BENCHMARKS: yes
BUILD_BENCHMARKS_NONIUS: no
WITH_GMP: yes
GMP_INCLUDE_DIRS: /opt/csw/include
GMP_LIBRARIES: /opt/csw/lib/libgmp.so
WITH_BFD: no
WITH_ECM: no
WITH_PRIMESIEVE: no
WITH_FLINT: no
WITH_ARB: no
WITH_MPFR: no
WITH_PIRANHA: no
WITH_LLVM: no
WITH_BOOST: no
WITH_PTHREAD: no
WITH_MPC: no
WITH_TCMALLOC: no
WITH_OPENMP: no
WITH_VIRTUAL_TYPEID: no

--------------------------------------------------------------------------------

-- Configuring done
-- Generating done
-- Build files have been written to: /symengine
```


make output:

```
[  0%] Generating CXX unity source symengine/cotire/symengine_CXX_unity.cxx
[  1%] Generating CXX prefix source symengine/cotire/symengine_CXX_prefix.cxx
[  1%] Generating CXX prefix header symengine/cotire/symengine_CXX_prefix.hxx
[  2%] Building CXX precompiled header symengine/cotire/symengine_CXX_prefix.hxx.gch
Scanning dependencies of target symengine
[  3%] Building CXX object symengine/CMakeFiles/symengine.dir/add.cpp.o
In file included from /symengine/symengine/add.h:10:0,
                 from /symengine/symengine/add.cpp:1:
/symengine/symengine/basic.h: In member function 'std::size_t SymEngine::RCPBasicHash::operator()(const SymEngine::RCP<const SymEngine::Basic>&) const':
/symengine/symengine/basic.h:196:24: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return k->hash();
                        ^
In file included from /symengine/symengine/basic.h:305:0,
                 from /symengine/symengine/add.h:10,
                 from /symengine/symengine/add.cpp:1:
/symengine/symengine/basic-inl.h: In member function 'std::size_t std::hash<SymEngine::Basic>::operator()(const SymEngine::Basic&) const':
/symengine/symengine/basic-inl.h:132:23: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return b.hash();
                       ^
[  3%] Building CXX object symengine/CMakeFiles/symengine.dir/basic.cpp.o
In file included from /symengine/symengine/printers.h:4:0,
                 from /symengine/symengine/basic.cpp:1:
/symengine/symengine/basic.h: In member function 'std::size_t SymEngine::RCPBasicHash::operator()(const SymEngine::RCP<const SymEngine::Basic>&) const':
/symengine/symengine/basic.h:196:24: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return k->hash();
                        ^
In file included from /symengine/symengine/basic.h:305:0,
                 from /symengine/symengine/printers.h:4,
                 from /symengine/symengine/basic.cpp:1:
/symengine/symengine/basic-inl.h: In member function 'std::size_t std::hash<SymEngine::Basic>::operator()(const SymEngine::Basic&) const':
/symengine/symengine/basic-inl.h:132:23: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return b.hash();
                       ^
[  4%] Building CXX object symengine/CMakeFiles/symengine.dir/complex.cpp.o
In file included from /symengine/symengine/number.h:10:0,
                 from /symengine/symengine/integer.h:10,
                 from /symengine/symengine/constants.h:10,
                 from /symengine/symengine/rational.h:9,
                 from /symengine/symengine/complex.h:9,
                 from /symengine/symengine/complex.cpp:1:
/symengine/symengine/basic.h: In member function 'std::size_t SymEngine::RCPBasicHash::operator()(const SymEngine::RCP<const SymEngine::Basic>&) const':
/symengine/symengine/basic.h:196:24: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return k->hash();
                        ^
In file included from /symengine/symengine/basic.h:305:0,
                 from /symengine/symengine/number.h:10,
                 from /symengine/symengine/integer.h:10,
                 from /symengine/symengine/constants.h:10,
                 from /symengine/symengine/rational.h:9,
                 from /symengine/symengine/complex.h:9,
                 from /symengine/symengine/complex.cpp:1:
/symengine/symengine/basic-inl.h: In member function 'std::size_t std::hash<SymEngine::Basic>::operator()(const SymEngine::Basic&) const':
/symengine/symengine/basic-inl.h:132:23: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return b.hash();
                       ^
[  4%] Building CXX object symengine/CMakeFiles/symengine.dir/complex_double.cpp.o
In file included from /symengine/symengine/number.h:10:0,
                 from /symengine/symengine/integer.h:10,
                 from /symengine/symengine/constants.h:10,
                 from /symengine/symengine/rational.h:9,
                 from /symengine/symengine/complex.h:9,
                 from /symengine/symengine/real_double.h:9,
                 from /symengine/symengine/complex_double.h:9,
                 from /symengine/symengine/complex_double.cpp:6:
/symengine/symengine/basic.h: In member function 'std::size_t SymEngine::RCPBasicHash::operator()(const SymEngine::RCP<const SymEngine::Basic>&) const':
/symengine/symengine/basic.h:196:24: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return k->hash();
                        ^
In file included from /symengine/symengine/basic.h:305:0,
                 from /symengine/symengine/number.h:10,
                 from /symengine/symengine/integer.h:10,
                 from /symengine/symengine/constants.h:10,
                 from /symengine/symengine/rational.h:9,
                 from /symengine/symengine/complex.h:9,
                 from /symengine/symengine/real_double.h:9,
                 from /symengine/symengine/complex_double.h:9,
                 from /symengine/symengine/complex_double.cpp:6:
/symengine/symengine/basic-inl.h: In member function 'std::size_t std::hash<SymEngine::Basic>::operator()(const SymEngine::Basic&) const':
/symengine/symengine/basic-inl.h:132:23: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return b.hash();
                       ^
[  5%] Building CXX object symengine/CMakeFiles/symengine.dir/constants.cpp.o
In file included from /symengine/symengine/number.h:10:0,
                 from /symengine/symengine/integer.h:10,
                 from /symengine/symengine/constants.h:10,
                 from /symengine/symengine/rational.h:9,
                 from /symengine/symengine/complex.h:9,
                 from /symengine/symengine/constants.cpp:1:
/symengine/symengine/basic.h: In member function 'std::size_t SymEngine::RCPBasicHash::operator()(const SymEngine::RCP<const SymEngine::Basic>&) const':
/symengine/symengine/basic.h:196:24: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return k->hash();
                        ^
In file included from /symengine/symengine/basic.h:305:0,
                 from /symengine/symengine/number.h:10,
                 from /symengine/symengine/integer.h:10,
                 from /symengine/symengine/constants.h:10,
                 from /symengine/symengine/rational.h:9,
                 from /symengine/symengine/complex.h:9,
                 from /symengine/symengine/constants.cpp:1:
/symengine/symengine/basic-inl.h: In member function 'std::size_t std::hash<SymEngine::Basic>::operator()(const SymEngine::Basic&) const':
/symengine/symengine/basic-inl.h:132:23: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return b.hash();
                       ^
[  5%] Building CXX object symengine/CMakeFiles/symengine.dir/cse.cpp.o
In file included from /symengine/symengine/cse.cpp:1:0:
/symengine/symengine/basic.h: In member function 'std::size_t SymEngine::RCPBasicHash::operator()(const SymEngine::RCP<const SymEngine::Basic>&) const':
/symengine/symengine/basic.h:196:24: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return k->hash();
                        ^
In file included from /symengine/symengine/basic.h:305:0,
                 from /symengine/symengine/cse.cpp:1:
/symengine/symengine/basic-inl.h: In member function 'std::size_t std::hash<SymEngine::Basic>::operator()(const SymEngine::Basic&) const':
/symengine/symengine/basic-inl.h:132:23: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return b.hash();
                       ^
[  6%] Building CXX object symengine/CMakeFiles/symengine.dir/cwrapper.cpp.o
In file included from /symengine/symengine/symbol.h:9:0,
                 from /symengine/symengine/cwrapper.cpp:4:
/symengine/symengine/basic.h: In member function 'std::size_t SymEngine::RCPBasicHash::operator()(const SymEngine::RCP<const SymEngine::Basic>&) const':
/symengine/symengine/basic.h:196:24: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return k->hash();
                        ^
In file included from /symengine/symengine/basic.h:305:0,
                 from /symengine/symengine/symbol.h:9,
                 from /symengine/symengine/cwrapper.cpp:4:
/symengine/symengine/basic-inl.h: In member function 'std::size_t std::hash<SymEngine::Basic>::operator()(const SymEngine::Basic&) const':
/symengine/symengine/basic-inl.h:132:23: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return b.hash();
                       ^
/symengine/symengine/cwrapper.cpp: In function 'std::size_t basic_hash(const basic_struct*)':
/symengine/symengine/cwrapper.cpp:1347:26: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
     return self->m->hash();
                          ^
[  6%] Building CXX object symengine/CMakeFiles/symengine.dir/dense_matrix.cpp.o
In file included from /symengine/symengine/matrix.h:4:0,
                 from /symengine/symengine/dense_matrix.cpp:1:
/symengine/symengine/basic.h: In member function 'std::size_t SymEngine::RCPBasicHash::operator()(const SymEngine::RCP<const SymEngine::Basic>&) const':
/symengine/symengine/basic.h:196:24: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return k->hash();
                        ^
In file included from /symengine/symengine/basic.h:305:0,
                 from /symengine/symengine/matrix.h:4,
                 from /symengine/symengine/dense_matrix.cpp:1:
/symengine/symengine/basic-inl.h: In member function 'std::size_t std::hash<SymEngine::Basic>::operator()(const SymEngine::Basic&) const':
/symengine/symengine/basic-inl.h:132:23: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return b.hash();
                       ^
[  6%] Building CXX object symengine/CMakeFiles/symengine.dir/derivative.cpp.o
In file included from /symengine/symengine/polys/upolybase.h:8:0,
                 from /symengine/symengine/polys/uintpoly_flint.h:8,
                 from /symengine/symengine/visitor.h:9,
                 from /symengine/symengine/derivative.cpp:1:
/symengine/symengine/basic.h: In member function 'std::size_t SymEngine::RCPBasicHash::operator()(const SymEngine::RCP<const SymEngine::Basic>&) const':
/symengine/symengine/basic.h:196:24: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return k->hash();
                        ^
In file included from /symengine/symengine/basic.h:305:0,
                 from /symengine/symengine/polys/upolybase.h:8,
                 from /symengine/symengine/polys/uintpoly_flint.h:8,
                 from /symengine/symengine/visitor.h:9,
                 from /symengine/symengine/derivative.cpp:1:
/symengine/symengine/basic-inl.h: In member function 'std::size_t std::hash<SymEngine::Basic>::operator()(const SymEngine::Basic&) const':
/symengine/symengine/basic-inl.h:132:23: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return b.hash();
                       ^
[  7%] Building CXX object symengine/CMakeFiles/symengine.dir/dict.cpp.o
In file included from /symengine/symengine/add.h:10:0,
                 from /symengine/symengine/expression.h:10,
                 from /symengine/symengine/dict.cpp:1:
/symengine/symengine/basic.h: In member function 'std::size_t SymEngine::RCPBasicHash::operator()(const SymEngine::RCP<const SymEngine::Basic>&) const':
/symengine/symengine/basic.h:196:24: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return k->hash();
                        ^
In file included from /symengine/symengine/basic.h:305:0,
                 from /symengine/symengine/add.h:10,
                 from /symengine/symengine/expression.h:10,
                 from /symengine/symengine/dict.cpp:1:
/symengine/symengine/basic-inl.h: In member function 'std::size_t std::hash<SymEngine::Basic>::operator()(const SymEngine::Basic&) const':
/symengine/symengine/basic-inl.h:132:23: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return b.hash();
                       ^
[  7%] Building CXX object symengine/CMakeFiles/symengine.dir/diophantine.cpp.o
In file included from /symengine/symengine/matrix.h:4:0,
                 from /symengine/symengine/diophantine.h:10,
                 from /symengine/symengine/diophantine.cpp:1:
/symengine/symengine/basic.h: In member function 'std::size_t SymEngine::RCPBasicHash::operator()(const SymEngine::RCP<const SymEngine::Basic>&) const':
/symengine/symengine/basic.h:196:24: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return k->hash();
                        ^
In file included from /symengine/symengine/basic.h:305:0,
                 from /symengine/symengine/matrix.h:4,
                 from /symengine/symengine/diophantine.h:10,
                 from /symengine/symengine/diophantine.cpp:1:
/symengine/symengine/basic-inl.h: In member function 'std::size_t std::hash<SymEngine::Basic>::operator()(const SymEngine::Basic&) const':
/symengine/symengine/basic-inl.h:132:23: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return b.hash();
                       ^
[  8%] Building CXX object symengine/CMakeFiles/symengine.dir/eval.cpp.o
In file included from /symengine/symengine/eval.h:8:0,
                 from /symengine/symengine/eval.cpp:2:
/symengine/symengine/basic.h: In member function 'std::size_t SymEngine::RCPBasicHash::operator()(const SymEngine::RCP<const SymEngine::Basic>&) const':
/symengine/symengine/basic.h:196:24: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return k->hash();
                        ^
In file included from /symengine/symengine/basic.h:305:0,
                 from /symengine/symengine/eval.h:8,
                 from /symengine/symengine/eval.cpp:2:
/symengine/symengine/basic-inl.h: In member function 'std::size_t std::hash<SymEngine::Basic>::operator()(const SymEngine::Basic&) const':
/symengine/symengine/basic-inl.h:132:23: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return b.hash();
                       ^
[  8%] Building CXX object symengine/CMakeFiles/symengine.dir/eval_double.cpp.o
In file included from /symengine/symengine/polys/upolybase.h:8:0,
                 from /symengine/symengine/polys/uintpoly_flint.h:8,
                 from /symengine/symengine/visitor.h:9,
                 from /symengine/symengine/eval_double.cpp:1:
/symengine/symengine/basic.h: In member function 'std::size_t SymEngine::RCPBasicHash::operator()(const SymEngine::RCP<const SymEngine::Basic>&) const':
/symengine/symengine/basic.h:196:24: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return k->hash();
                        ^
In file included from /symengine/symengine/basic.h:305:0,
                 from /symengine/symengine/polys/upolybase.h:8,
                 from /symengine/symengine/polys/uintpoly_flint.h:8,
                 from /symengine/symengine/visitor.h:9,
                 from /symengine/symengine/eval_double.cpp:1:
/symengine/symengine/basic-inl.h: In member function 'std::size_t std::hash<SymEngine::Basic>::operator()(const SymEngine::Basic&) const':
/symengine/symengine/basic-inl.h:132:23: warning: conversion to 'std::size_t {aka unsigned int}' from 'SymEngine::hash_t {aka long long unsigned int}' may alter its value [-Wconversion]
         return b.hash();
                       ^
/symengine/symengine/eval_double.cpp: In lambda function:
/symengine/symengine/eval_double.cpp:681:33: error: call of overloaded 'atan2(int, int)' is ambiguous
             return ::atan2(0, -1);
                                 ^
In file included from /usr/include/math.h:11:0,
                 from /opt/csw/include/c++/5.2.0/cmath:44,
                 from /symengine/symengine/cotire/symengine_CXX_prefix.cxx:13,
                 from /symengine/symengine/cotire/symengine_CXX_prefix.hxx:4:
/opt/csw/lib/gcc/i386-pc-solaris2.10/5.2.0/include-fixed/iso/math_iso.h:186:21: note: candidate: long double std::atan2(long double, long double)
  inline long double atan2(long double __X, long double __Y) { return
                     ^
/opt/csw/lib/gcc/i386-pc-solaris2.10/5.2.0/include-fixed/iso/math_iso.h:158:15: note: candidate: float std::atan2(float, float)
  inline float atan2(float __X, float __Y) { return __atan2f(__X, __Y); }
               ^
/opt/csw/lib/gcc/i386-pc-solaris2.10/5.2.0/include-fixed/iso/math_iso.h:56:15: note: candidate: double std::atan2(double, double)
 extern double atan2 __P((double, double));
               ^
/symengine/symengine/eval_double.cpp:683:27: error: call of overloaded 'exp(int)' is ambiguous
             return ::exp(1);
                           ^
In file included from /usr/include/math.h:11:0,
                 from /opt/csw/include/c++/5.2.0/cmath:44,
                 from /symengine/symengine/cotire/symengine_CXX_prefix.cxx:13,
                 from /symengine/symengine/cotire/symengine_CXX_prefix.hxx:4:
/opt/csw/lib/gcc/i386-pc-solaris2.10/5.2.0/include-fixed/iso/math_iso.h:191:21: note: candidate: long double std::exp(long double)
  inline long double exp(long double __X) { return __expl(__X); }
                     ^
/opt/csw/lib/gcc/i386-pc-solaris2.10/5.2.0/include-fixed/iso/math_iso.h:162:15: note: candidate: float std::exp(float)
  inline float exp(float __X) { return __expf(__X); }
               ^
/opt/csw/lib/gcc/i386-pc-solaris2.10/5.2.0/include-fixed/iso/math_iso.h:65:15: note: candidate: double std::exp(double)
 extern double exp __P((double));
               ^
*** Error code 1
The following command caused the error:
cd /symengine/symengine && /opt/csw/bin/g++  -Dsymengine_EXPORTS -I/symengine -std=c++11 -O2 -fPIC -Wall -Wextra -Wno-unused-parameter -fno-common -O3 -funroll-loops -Wno-unknown-pragmas   -Wconversion -Wno-sign-conversion -Winvalid-pch -include /symengine/symengine/cotire/symengine_CXX_prefix.hxx -o CMakeFiles/symengine.dir/eval_double.cpp.o -c /symengine/symengine/eval_double.cpp
make: Fatal error: Command failed for target `symengine/CMakeFiles/symengine.dir/eval_double.cpp.o'
Current working directory /symengine
*** Error code 1
The following command caused the error:
make -f symengine/CMakeFiles/symengine.dir/build.make symengine/CMakeFiles/symengine.dir/build
make: Fatal error: Command failed for target `symengine/CMakeFiles/symengine.dir/all'
Current working directory /symengine
*** Error code 1
The following command caused the error:
make -f CMakeFiles/Makefile2 all
make: Fatal error: Command failed for target `all'
```

--- a/symengine/eval_double.cpp
+++ b/symengine/eval_double.cpp
@@ -678,9 +678,9 @@ std::vector<fn> init_eval_double()
     };
     table[SYMENGINE_CONSTANT] = [](const Basic &x) {
         if (eq(x, *pi)) {
-            return ::atan2(0, -1);
+            return std::atan2(0, -1);
         } else if (eq(x, *E)) {
-            return ::exp(1);
+            return std::exp(1);
         } else if (eq(x, *EulerGamma)) {
             return 0.5772156649015328606065; // use until polygamma or digamma
                                              // is implemented

--- a/symengine/tests/basic/test_parser.cpp
+++ b/symengine/tests/basic/test_parser.cpp
@@ -629,7 +629,7 @@ TEST_CASE("Parsing: doubles", "[parser]")
     res = parse(s);
     REQUIRE(is_a<RealDouble>(*res));
     d = down_cast<const RealDouble &>(*res).as_double();
-    REQUIRE(std::abs(d - (::sqrt(2) + 5)) < 1e-12);
+    REQUIRE(std::abs(d - (std::sqrt(2) + 5)) < 1e-12);
 
     // Test that https://github.com/symengine/symengine/issues/1413 is fixed
 

